it/controllers/incomeSources/add/AddBusinessTradeControllerISpec.scala

```diff
@@ -0,0 +1,177 @@
+          )
+        }
+
+        When(s"I call POST ${addBusinessTradeSubmitUrl}")
+        val result = IncomeTaxViewChangeFrontend.post("/income-sources/add/business-trade")(formData)
+        result should have(
+          httpStatus(SEE_OTHER),
+          redirectURI(addBusinessAddressUrl)
+        )
+      }
+    }
+    "show error when form is filled incorrectly" in {
+      enable(IncomeSources)
+      IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+      val formData: Map[String, Seq[String]] = {
+        Map(
+          BusinessTradeForm.businessTrade -> Seq("")
+        )
+      }
+
+      val result = IncomeTaxViewChangeFrontend.post("/income-sources/add/business-trade")(formData)
+      result should have(
+        httpStatus(BAD_REQUEST),
+        elementTextByID("business-trade-error")(messagesAPI("base.error-prefix") + " " +
+          messagesAPI("add-business-trade.form.error.empty"))
+      )
+    }
+  }
+
+
+  s"calling GET $changeBusinessTradeUrl" should {
+    "render the Change Business Trade page" when {
+      "User is authorised" in {
+        Given("I wiremock stub a successful Income Source Details response with no businesses or properties")
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesResponse)
+
+        When(s"I call GET $changeBusinessTradeUrl")
+        val result = IncomeTaxViewChangeFrontend.get("/income-sources/add/change-business-trade")
+
+        result should have(
+          httpStatus(OK),
+          pageTitleIndividual(pageTitleMsgKey),
+          elementTextByID("business-trade-hint")(pageHint),
+          elementTextByID("continue-button")(button)
+        )
+      }
+    }
+    "303 SEE_OTHER - redirect to home page" when {
+      "Income Sources FS disabled" in {
+        Given("I wiremock stub a successful Income Source Details response with no businesses or properties")
+        disable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        When(s"I call GET ${changeBusinessTradeUrl}")
+        val result = IncomeTaxViewChangeFrontend.get("/income-sources/add/change-business-trade")
+        result should have(
+          httpStatus(SEE_OTHER),
+          redirectURI(incomeSourcesUrl)
+        )
+      }
+    }
+  }
+
+  s"calling POST ${submitChangeBusinessTradeUrl}" should {
+    s"303 SEE_OTHER and redirect to $addBusinessAddressUrl" when {
+      "User is authorised and business trade is valid" in {
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        val formData: Map[String, Seq[String]] = {
+          Map(
+            BusinessTradeForm.businessTrade -> Seq("Test Business Trade")
+          )
+        }
+
```

Comment:
Can you add a check for business trade name ? check with database using the session service

also you need to set the business name in session and Trade name